networks/xml_cnn.py

```diff
@@ -0,0 +1,71 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from networks.base import BaseModel
+
+
+def out_size(l_in, kernel_size, channels, padding=0, dilation=1, stride=1):
+    a = l_in + 2 * padding - dilation * (kernel_size - 1) - 1
+    b = int(a / stride)
+    return (b + 1) * channels
+
+
+class XMLCNN(BaseModel):
+    def __init__(self, config, embed_vecs):
+        super(XMLCNN, self).__init__(config, embed_vecs)
+        assert config.fixed_length is True
+
+        self.filter_sizes = config.filter_sizes
+        emb_dim = embed_vecs.shape[1]
+        num_filter_maps = config.num_filter_maps
+        strides = config.strides
+        d_max_pool_p = config.d_max_pool_p
+
+        self.convs = nn.ModuleList()
+        self.poolings = nn.ModuleList()
+
+        for filter_size, p, stride in zip(self.filter_sizes, d_max_pool_p, strides):
+            conv = nn.Conv2d(
+                in_channels=1,
+                out_channels=num_filter_maps,
+                kernel_size=(filter_size, emb_dim),
+                stride=(stride, emb_dim))
```

Comment:
This can is indeed a Conv1d operation.. Not sure if we want to do 2d here?? Need some discussion I think